src/ColorSchemeToggle/AutoIcon.tsx

```diff
@@ -0,0 +1,71 @@
+import { type FC } from "react";
+import { type ColorScheme } from "../ColorContext";
+
+interface AutoIconProps extends React.SVGAttributes<SVGElement> {
+  colorScheme: ColorScheme;
+}
+
+const AutoIconLight: FC<React.SVGAttributes<SVGElement>> = (props) => (
+  <AutoIcon colorScheme="light" {...props} />
+);
+const AutoIconDark: FC<React.SVGAttributes<SVGElement>> = (props) => (
+  <AutoIcon colorScheme="dark" {...props} />
+);
+
+const AutoIcon: FC<AutoIconProps> = ({ colorScheme, ...props }) => (
+  <svg
+    {...props}
```

Comment:
We don't actually need to pass the scheme to `<AutoIcon />` ; it can just read it from context.

I've only included the start of the suggestion. Most of the rest of it is just reindenting.

```suggestion
const AutoIcon: FC<React.SVGAttributes<SVGElement>> = props => {
  const { scheme } = useContext(ColorContext);

  return (
    <svg
      {...props}
      aria-label={scheme === 'light' ? "Light mode (automatic)" : "Dark mode (automatic)"}
```